options.go

```diff
@@ -82,10 +86,9 @@ func WithEndpointErrorHandler(eh EndpointErrorHandler) EndpointOption {
 
 type withUpdatePeriod struct{ updatePeriod time.Duration }
 
-func (o withUpdatePeriod) applyProtocol(p *Protocol) { p.updatePeriod = o.updatePeriod }
-func (o withUpdatePeriod) applyEndpoint(e *Endpoint) { e.updatePeriod = o.updatePeriod }
+func (o withUpdatePeriod) applyConn(c *Conn) { c.updatePeriod = o.updatePeriod }
 
-func WithUpdatePeriod(updatePeriod time.Duration) Option {
+func WithUpdatePeriod(updatePeriod time.Duration) ConnOption {
```

Comment:
warning: exported function WithUpdatePeriod should have comment or be unexported (golint)